snap/hooks/configure

```diff
@@ -0,0 +1,67 @@
+[ -z "$(snapctl get disable-service.container-infra)" ] && snapctl set disable-service.container-infra=false
+[ -z "$(snapctl get disable-service.database)" ] && snapctl set disable-service.database=false
+[ -z "$(snapctl get disable-service.dns)" ] && snapctl set disable-service.dns=false
+[ -z "$(snapctl get disable-service.gnocchi)" ] && snapctl set disable-service.gnocchi=false
+[ -z "$(snapctl get disable-service.identity)" ] && snapctl set disable-service.identity=false
+[ -z "$(snapctl get disable-service.image)" ] && snapctl set disable-service.image=false
+[ -z "$(snapctl get disable-service.load-balancer)" ] && snapctl set disable-service.load-balancer=false
+[ -z "$(snapctl get disable-service.network)" ] && snapctl set disable-service.network=false
+[ -z "$(snapctl get disable-service.object-store)" ] && snapctl set disable-service.object-store=false
+[ -z "$(snapctl get disable-service.orchestration)" ] && snapctl set disable-service.orchestration=false
+[ -z "$(snapctl get disable-service.placement)" ] && snapctl set disable-service.placement=false
+[ -z "$(snapctl get disable-service.volume)" ] && snapctl set disable-service.volume=false
+[ -z "$(snapctl get disable-slow-metrics)" ] && snapctl set disable-slow-metrics=false
+[ -z "$(snapctl get domain-id)" ] && snapctl set domain-id=
+[ -z "$(snapctl get endpoint-type)" ] && snapctl set endpoint-type=
+[ -z "$(snapctl get log.format)" ] && snapctl set log.format=
+[ -z "$(snapctl get log.level)" ] && snapctl set log.level=
+[ -z "$(snapctl get multi-cloud)" ] && snapctl set multi-cloud=false
+[ -z "$(snapctl get os-client-config)" ] && snapctl set os-client-config=
+[ -z "$(snapctl get prefix)" ] && snapctl set prefix=
+[ -z "$(snapctl get web.listen-address)" ] && snapctl set web.listen-address=
+[ -z "$(snapctl get web.telemetry-path)" ] && snapctl set web.telemetry-path=
+
+validate_bool() {
+    key="$1"
+    value="$(snapctl get $key)"
+    if [ "$value" != "true" -a "$value" != "false" ]
+    then
+        echo "\"$key\" is not a valid boolean value (must be \"true\" or \"false\")" >&2
+        exit 1
+    fi
+}
+
+validate_bool collect-metric-time
+validate_bool disable-cinder-agent-uuid
+validate_bool disable-deprecated-metrics
+validate_bool disable-service.baremetal
+validate_bool disable-service.compute
+validate_bool disable-service.container-infra
+validate_bool disable-service.database
+validate_bool disable-service.dns
+validate_bool disable-service.gnocchi
+validate_bool disable-service.identity
+validate_bool disable-service.image
+validate_bool disable-service.load-balancer
+validate_bool disable-service.network
+validate_bool disable-service.object-store
+validate_bool disable-service.orchestration
+validate_bool disable-service.placement
+validate_bool disable-service.volume
+validate_bool disable-slow-metrics
+validate_bool multi-cloud
+
+# Restart openstack-exporter service to apply new config.
+snapctl restart $SNAP_NAME
```

Comment:
I didn't notice that the service is now disable on install. I think some return message would be helpful (if it's possible). So it's like:

1. if the service is not started, running `snap set / unset` should print out `Warning: the service is not started, please start the service to apply the changes`
2. if the service is already running, `snap set / unset` should not print out anything

some thing like that

Reply:
Not quite, it's technically possible to exit with non zero status and echo some message, like you did with the `validate_bool` function. But it's going to be unnecessarily complex to get the status of the snap service though. 

Yeah, I agree now we should not restart it given that the snap service is not started by default. (otherwise, I think we should restart, because it provide better user experience.)

Reply:
> It seems that messages are only shown if the exit status is not success

Yes, that's what I meant, we can `exit 1` for example

> but we don't want to do that because then it won't save the config

Interesting, I did not expect that, the config should be set before restarting. Maybe you are thinking exit immediately when the service is not started and configure hook is called?

But anyways, that's irrelevant now

Reply:
Okay, thanks! Learn something new